### PR TITLE
add missing pkg-config builddep to Guile

### PR DESCRIPTION
**Pull request by fizwit**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/g/Guile/Guile-3.0.7-GCCcore-11.2.0.eb [hunk 1]
```diff
@@ -21,6 +21,7 @@ checksums = ['c7935b7a29e42443f6a35d35cf20ffa7d028c399303f872cd1219598a83656ae']
 builddependencies = [
     ('Autotools', '20210726'),
     ('binutils', '2.37'),
+    ('pkg-config', '0.29.2'),
```
Comment (smoors) [inline easybuild/easyconfigs/g/Guile/Guile-3.0.7-GCCcore-11.2.0.eb:24]: can we use `pkgconf` here instead?

Comment (fizwit) [reply easybuild/easyconfigs/g/Guile/Guile-3.0.7-GCCcore-11.2.0.eb:24]: Yes, pkgconf is correct.

Comment (fizwit) [reply easybuild/easyconfigs/g/Guile/Guile-3.0.7-GCCcore-11.2.0.eb]: pkg-conf was correct.

Comment (smoors) [reply easybuild/easyconfigs/g/Guile/Guile-3.0.7-GCCcore-11.2.0.eb]: I meant to use `pkgconf` as a drop-in replacement for `pkg-config`.
I went ahead and made the change for you if you don't mind.

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 3 out of 3 (1 easyconfigs in total)
node301.hydra.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/a474510e80cc2b24179ae874c56a1e84 for a full test report.

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @fizwit!

**Pull request merged**